src/main/java/com/griddynamics/cloud/learning/service/ArticleService.java

```diff
@@ -77,4 +91,34 @@ public ArticleService(ArticleRepository repository, ArticleMapper articleMapper)
                 .map(articleMapper::articleToArticleDto)
                 .collect(Collectors.toList()), articles.getPageable(), articles.getTotalElements());
     }
+
+    public ArticleLikeDto likeAnArticle(final ArticleLikeDto likeDto) {
+
+        final String currentUserUsername = getCurrentUserUsername();
+        final Long currentArticleId = likeDto.getArticleId();
+        final Boolean liked = likeDto.getLiked();
+
+        final UserArticle userArticle = StreamSupport.stream(userArticleRepository.findAll().spliterator(), Boolean.FALSE)
```

Comment:
don't query ALL the table please, make a dedicate DAO API for retreiving a user-article entity by user id and article id